### PR TITLE
Support new os SNR

### PR DESCRIPTION
**Pull request by hrtrd**
mib file

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
You also need to add more files to support a new OS, you can read about it here; https://docs.librenms.org/Developing/Support-New-OS/

**Comment by hrtrd** [comment]
> You also need to add more files to support a new OS, you can read about it here; https://docs.librenms.org/Developing/Support-New-OS/

yeah, all ready and work on my prod. Add today

**Comment by hrtrd** [comment]
@Jellyfrog why some errors? Logo failed?

**Comment by hrtrd** [comment]
@Jellyfrog hm, try this ./scripts/collect-snmp-data.php -h hostname, and got error:

In snmp.inc.php line 270:                                           
  Call to undefined function report_this_text()

**Comment by Jellyfrog** [comment]
> @Jellyfrog hm, try this ./scripts/collect-snmp-data.php -h hostname, and got error:
> 
> In snmp.inc.php line 270:
> Call to undefined function report_this_text()

Seems like a bug in the code,
please change file `/includes/snmp.inc.php` row 270:
```diff
-echo report_this_text("snmp_get called for multiple OIDs: $oid");
+echo "snmp_get called for multiple OIDs: $oid";
```

**Comment by hrtrd** [comment]
> > @Jellyfrog hm, try this ./scripts/collect-snmp-data.php -h hostname, and got error:
> > In snmp.inc.php line 270:
> > Call to undefined function report_this_text()
> 
> Seems like a bug in the code,
> please change file `/includes/snmp.inc.php` row 270:
> 
> ```diff
> -echo report_this_text("snmp_get called for multiple OIDs: $oid");
> +echo "snmp_get called for multiple OIDs: $oid";
> ```

yes, its fixed ./scripts/collect-snmp-data.php -h hostname finished.

next step ./scripts/save-test-data.php -o snr, and got another error:
./scripts/save-test-data.php -o snr
Starting snmpsim listening on 127.1.6.1:1161... 
OS: snr
Module: all

snr: SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost' (SQL: select * from `ipv4_addresses` where `ipv4_address` = 127.1.6.1 limit 1)

File: includes/definitions/discovery/snr.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                    descr: 'Temperature {{ $index }}'
+        state:
+           data:
+                -
+                    oid: sysFanSpeed
+                    num_oid: '.1.3.6.1.4.1.40418.7.100.1.12.1.4.{{ $index }}'
+                    descr: 'Fan {{ $index }}'
+                -
+                    oid: showSwitchStartUpStatus
+                    num_oid: ' .1.3.6.1.4.1.40418.7.100.1.16.{{ $index }}'
```
Comment (murrant) [inline includes/definitions/discovery/snr.yaml]: there is a leading space here.

Comment (hrtrd) [reply includes/definitions/discovery/snr.yaml]: fixed

**Comment by murrant** [comment]
@hrtrd I don't have permission to edit this PR, so here is the json.


[snr.zip](https://github.com/librenms/librenms/files/3748012/snr.zip)

File: includes/snmp.inc.php [hunk 2]
```diff
@@ -267,7 +267,7 @@ function snmp_get($device, $oid, $options = null, $mib = null, $mibdir = null)
     $time_start = microtime(true);
 
     if (strstr($oid, ' ')) {
-        echo report_this_text("snmp_get called for multiple OIDs: $oid");
+        echo "snmp_get called for multiple OIDs: $oid";
```
Comment (murrant) [inline includes/snmp.inc.php]: This should be `throw new Exception("snmp_get called for multiple OIDs: $oid")`

File: tests/snmpsim/snr.snmprec [hunk 3]
```diff
@@ -0,0 +1,1261 @@
+ All rights reserved
```
Comment (murrant) [inline tests/snmpsim/snr.snmprec]: It looks like these are encoded incorrectly...  Delete these text lines.
I assume they are part of the sysDescr but got messed up somehow.
If you send me the output of snmpget 1.3.6.1.2.1.1.1.0 from the device, I'll encode it correctly.

File: includes/snmp.inc.php (hunk 2 above)
Comment (Jellyfrog) [reply includes/snmp.inc.php]: This was fixed in https://github.com/librenms/librenms/pull/10728

File: includes/definitions/discovery/snr.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    sensors:
+        temperature:
+           data:
+                -
+                    oid: sysTemperature
+                    num_oid: '.1.3.6.1.4.1.40418.7.100.1.8.1.10.{{ $index }}'
+                    descr: 'Temperature {{ $index }}'
+        state:
+           data:
+                -
```
Comment (murrant) [inline includes/definitions/discovery/snr.yaml]: The states field is required for state sensors.

**Comment by hrtrd** [comment]
> @hrtrd I don't have permission to edit this PR, so here is the json.
> 
> [snr.zip](https://github.com/librenms/librenms/files/3748012/snr.zip)

where to put it?

File: tests/snmpsim/snr.snmprec (hunk 3 above)
Comment (hrtrd) [reply tests/snmpsim/snr.snmprec]: 1.3.6.1.2.1.1.1.0
SNMPv2-MIB::sysDescr.0 = STRING: SNR-S2990X-24FQ Device, Compiled Jan 10 13:47:05 2019
 SoftWare Version 7.5.3.2(R0010.0044)
 BootRom Version 7.5.6
 HardWare Version 1.0.1
 Serial No.:SW080710J513000055
 Copyright (C) 2019 NAG LLC
 All rights reserved

File: includes/definitions/discovery/snr.yaml (hunk 4 above)
Comment (hrtrd) [reply includes/definitions/discovery/snr.yaml]: it is with this model that the only thing that can be assigned to the state is the fan status and whether the ROM download is successful

Comment (hrtrd) [reply includes/definitions/discovery/snr.yaml]: Screenshot_50

**Comment by PipoCanaja** [comment]
The tests files need to be put with the others tests files : 

* .snmprec goes into ~/tests/snmpsim
* .json goes into ~/tests/data

File: includes/definitions/discovery/snr.yaml (hunk 4 above)
Comment (murrant) [reply includes/definitions/discovery/snr.yaml]: then you need 1 and 0

Comment (murrant) [reply includes/definitions/discovery/snr.yaml]: @hrtrd I checked the code and if you don't define states, the sensor will ALWAYS show ok, no matter what the state is...  So you need to define states.

**Comment by hrtrd** [comment]
> The tests files need to be put with the others tests files :
> 
> * .snmprec goes into ~/tests/snmpsim
> * .json goes into ~/tests/data

Screenshot_549
same error

**Comment by hrtrd** [comment]
./scripts/save-test-data.php -o snr
Starting snmpsim listening on 127.1.6.1:1161... 
OS: snr
Module: all

snr: Could not connect to 127.1.6.1, please check the snmp details and snmp reachability

**Comment by murrant** [comment]
@hrtrd something is trying to run an snmp_get with an empty oid...  I didn't notice where with a quick review.

**Comment by PipoCanaja** [comment]
Thanx @hrtrd

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-59-release-changelog-december-2019/10558/1

**Pull request merged**